components/server/ee/src/user/stripe-service.ts

```diff
@@ -149,11 +180,27 @@ export class StripeService {
         }
         const startOfNextMonth = new Date(new Date().toISOString().slice(0, 7) + "-01"); // First day of this month (YYYY-MM-01)
         startOfNextMonth.setMonth(startOfNextMonth.getMonth() + 1); // Add one month
-        await this.getStripe().subscriptions.create({
-            customer: customer.id,
-            items: [{ price: priceId }],
-            automatic_tax: { enabled: isAutomaticTaxSupported },
-            billing_cycle_anchor: Math.round(startOfNextMonth.getTime() / 1000),
+
+        await reportStripeOutcome("subscriptions_create", () => {
+            return this.getStripe().subscriptions.create({
+                customer: customer.id,
+                items: [{ price: priceId }],
+                automatic_tax: { enabled: isAutomaticTaxSupported },
+                billing_cycle_anchor: Math.round(startOfNextMonth.getTime() / 1000),
+            });
         });
     }
 }
+
+async function reportStripeOutcome<T>(op: string, f: () => T) {
```

Comment:
```suggestion
async function reportStripeOutcome<T>(op: string, f: () => Promise<T>): T {
```

Reply:
What's the difference? T can be either, no?

Reply:
I guess it forces the return to be a promise

Reply:
:+1: It avoids copy'n'paste errors where you insert an `await`, or forget what you're supposed to return.

Reply:
That makes sense, thanks for the tip!